File: 2270/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(waysToSplitArray([]int{10,4,-8,7}))
	fmt.Println(waysToSplitArray([]int{2,3,1,0}))
}

func waysToSplitArray(nums []int) int {
	res := 0
	l, r := 0, 0
	for i := range nums {
		r += nums[i]
	}
	for i := 0; i < len(nums)-1; i++ {
		l += nums[i]
		r -= nums[i]
		if l >= r {
			res++
		}
	}
	return res
}

File: 2270/main.rs
struct Solution {}

impl Solution {
    pub fn ways_to_split_array(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut left: i64 = 0;
        let mut right: i64 = nums.iter().map(|&x| x as i64).sum();
        let mut ans = 0;
        for i in 0..n-1 {
            left += nums[i] as i64;
            right -= nums[i] as i64;
            if left >= right {
                ans += 1;
            }
        }
        ans
    }
}

fn main() {
    println!("{:#?}", Solution::ways_to_split_array(vec![2,3,1,0]));
}